syntax = "proto3";
package database;

message DatabaseData {
    enum Operation {
        INSERT = 0;
        UPDATE = 1;
        DELETE = 2;
    }
    string _id = 6;
    Operation operation = 1;
    bool remote = 2;
    string ownership = 3;

    // Database data
    string imsi = 4;
    string security = 5;

    // For update only
    string update_data = 7;
}

// Used for testing / comparison
message OldDatabaseData {
    enum Operation {
        INSERT = 0;
        DELETE = 1;
        UPDATE = 2;
    }
    Operation operation = 1;
    string hex_encoded_object = 2;
}